packages/core/android/src/test/kotlin/com/datadog/tools/unit/forge/ResourceEventForgeryFactory.kt

```diff
@@ -0,0 +1,116 @@
+                    ResourceEvent.Provider(
+                        domain = aNullable { aStringMatching("[a-z]+\\.[a-z]{3}") },
+                        name = aNullable { anAlphabeticalString() },
+                        type = aNullable()
+                    )
+                }
+            ),
+            view = ResourceEvent.View(
+                id = forge.getForgery<UUID>().toString(),
+                url = forge.aStringMatching("https://[a-z]+.[a-z]{3}/[a-z0-9_/]+"),
+                referrer = forge.aNullable { getForgery<URL>().toString() },
+                name = forge.aNullable { anAlphabeticalString() }
+            ),
+            connectivity = forge.aNullable {
+                ResourceEvent.Connectivity(
+                    status = getForgery(),
+                    interfaces = aList { getForgery() },
+                    cellular = aNullable {
+                        ResourceEvent.Cellular(
+                            technology = aNullable { anAlphabeticalString() },
+                            carrierName = aNullable { anAlphabeticalString() }
+                        )
+                    }
+                )
+            },
+            synthetics = forge.aNullable {
+                ResourceEvent.Synthetics(
+                    testId = forge.anHexadecimalString(),
+                    resultId = forge.anHexadecimalString()
+                )
+            },
+            usr = forge.aNullable {
+                ResourceEvent.Usr(
+                    id = aNullable { anHexadecimalString() },
+                    name = aNullable { aStringMatching("[A-Z][a-z]+ [A-Z]\\. [A-Z][a-z]+") },
+                    email = aNullable { aStringMatching("[a-z]+\\.[a-z]+@[a-z]+\\.[a-z]{3}") },
+                    additionalProperties = mutableMapOf()
+                )
+            },
+            action = forge.aNullable {
+                ResourceEvent.Action(aList { getForgery<UUID>().toString() })
+            },
+            application = ResourceEvent.Application(forge.getForgery<UUID>().toString()),
+            service = forge.aNullable { anAlphabeticalString() },
+            session = ResourceEvent.ResourceEventSession(
+                id = forge.getForgery<UUID>().toString(),
+                type = ResourceEvent.ResourceEventSessionType.USER,
+                hasReplay = forge.aNullable { aBool() }
+            ),
+            source = forge.aNullable { aValueFrom(ResourceEvent.Source::class.java) },
+            ciTest = forge.aNullable {
+                ResourceEvent.CiTest(anHexadecimalString())
+            },
+            os = ResourceEvent.Os(
+                name = "android",
+                version = "11.3",
+                versionMajor = "11"
+            ),
+            device = ResourceEvent.Device(
+                name = "deviceName",
+                model = "model",
+                brand = "brand",
+                type = ResourceEvent.DeviceType.MOBILE,
```

Comment:
not something very important, probably, but for the string literals we can use `forge.a(....)String` here, and for enum `aValueFrom`.